app/src/main/java/com/adc/gpai/onboarding/ModifyTranscriptScreen.kt

```diff
@@ -1,117 +1,646 @@
+                            grade = grade
+                        )
+                        
+                        // Add to repository - now uncommented
+                        viewModel.addCourse(termId, newCourse)
+                        
+                        // Reset dialog state
+                        showAddCourseDialog = false
+                    }
+                }
+            )
+        }
+        
+        // Delete Confirmation Dialog
+        if (courseToDelete != null) {
+            AlertDialog(
+                onDismissRequest = { courseToDelete = null },
+                title = { Text("Confirm Deletion") },
+                text = { Text("Are you sure you want to delete this course: ${courseToDelete!!.course.courseName}?") },
+                confirmButton = {
+                    Button(
+                        onClick = {
+                            // Delete the course through the repository
+                            viewModel.removeCourse(courseToDelete!!.course)
+                            courseToDelete = null
+                        },
+                        colors = ButtonDefaults.buttonColors(containerColor = BrandDarkPurple)
+                    ) {
+                        Text("Delete")
+                    }
+                },
+                dismissButton = {
+                    TextButton(onClick = { courseToDelete = null }) {
+                        Text("Cancel")
+                    }
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun CourseItem(
+    course: Course,
+    onEdit: () -> Unit,
+    onDelete: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(Color.LightGray.copy(alpha = 0.2f), RoundedCornerShape(4.dp))
+            .padding(8.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
     ) {
-        Text(text = "Course List", style = MaterialTheme.typography.headlineLarge)
-        CourseList(transcript = transcript.value)
-        Button(onClick = { context.startActivity(Intent(context, HomeActivity::class.java)) }) {
-            Text(text = "Finish")
+        // Course name
+        Text(
+            text = "${course.courseCode} ${course.courseName}",
+            modifier = Modifier.weight(1f)
+        )
+        
+        // Credit hours
+        Text(
+            text = "${course.attempted} cr",
+            modifier = Modifier.padding(horizontal = 8.dp)
+        )
+        
+        // Grade
+        Surface(
+            shape = RoundedCornerShape(4.dp),
+            color = when (course.grade.firstOrNull() ?: '?') {
+                'A' -> Color(0xFF4CAF50) // Green
+                'B' -> Color(0xFF8BC34A) // Light Green
+                'C' -> Color(0xFFFFC107) // Yellow
+                'D' -> Color(0xFFFF9800) // Orange
+                else -> Color(0xFFF44336) // Red for F
```

Comment:
NC State has some other types of grades that can appear on a transcript: https://policies.ncsu.edu/regulation/reg-02-50-03/

I'd recommend at least adding a color for S so that it doesn't look like an F